web/src/components/Accordion.tsx

```diff
@@ -20,27 +20,20 @@ export default function Accordion({
   icon?: React.ReactNode;
   children?: React.ReactNode;
   title: string;
-  isCollapsedAtom?: PrimitiveAtom<boolean>;
+  isCollapsedAtom: PrimitiveAtom<boolean | undefined>;
 }) {
-  const [collapsedAtom, setCollapsedAtom] = isCollapsedAtom
-    ? useAtom(isCollapsedAtom)
-    : [null, null];
-  const [isCollapsed, setIsCollapsed] = useState(isCollapsedDefault);
-
+  const [isCollapsed, setIsCollapsed] = useAtom(isCollapsedAtom);
   useEffect(() => {
-    if (collapsedAtom !== null) {
-      setIsCollapsed(collapsedAtom);
+    if (isCollapsed === undefined) {
+      setIsCollapsed(isCollapsedDefault);
     }
-  }, [collapsedAtom]);
```

Comment:
I'm a bit unsure about these changes but it was the best way I could think of to resolve the fact that the atoms can't be conditionals.

Reply:
Not just the rule per se but react internally requires all hooks to be called in the same order on every render to function properly internally.

This means the calling of the atom itself can't be a conditional but you can choose to not use the atom later on.